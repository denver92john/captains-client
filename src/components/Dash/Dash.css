.dash {
    padding: 20px;
    background-color: ghostwhite;
    min-width: 200px;
    min-height: 200px;
    display: flex;
    flex-flow: row wrap;
    justify-content: space-evenly;
    align-items: center;
}

.dash-icon {
    flex-basis: 50%;
}

.dash-icon-img {
    width: 50px;
}

.dash-icon-text {
    font-size: 18px;
}