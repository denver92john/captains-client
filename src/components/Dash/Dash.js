import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import './Dash.css';

class Dash extends Component {
    render() {
        return (
            <div className="wrapper dash">
                <div className="dash-icon">
                    <Link to="/lists">
                        <FontAwesomeIcon icon="list" />
                    </Link>
                    <h2 className="dash-icon-text">Lists</h2>
                </div>
                <div className="dash-icon">
                    <Link to="/teams">
                        <FontAwesomeIcon icon="user-friends" />
                    </Link>
                    <h2 className="dash-icon-text">Teams</h2>
                </div>
            </div>
        );
    }
}

export default Dash;