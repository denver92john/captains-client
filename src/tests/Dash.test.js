import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import Dash from '../components/Dash/Dash';

describe('Dash component', () => {
    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <Dash />
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const tree = renderer
            .create(<BrowserRouter>
                    <Dash />
                </BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})