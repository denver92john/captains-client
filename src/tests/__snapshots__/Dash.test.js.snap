// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dash component renders UI as expected 1`] = `
<div
  className="wrapper dash"
>
  <div
    className="dash-icon"
  >
    <a
      href="/lists"
      onClick={[Function]}
    />
    <h2
      className="dash-icon-text"
    >
      Lists
    </h2>
  </div>
  <div
    className="dash-icon"
  >
    <a
      href="/teams"
      onClick={[Function]}
    />
    <h2
      className="dash-icon-text"
    >
      Teams
    </h2>
  </div>
</div>
`;
